visualization/app/codeCharta/ui/customConfigs/addCustomConfigButton/addCustomConfigDialog/addCustomConfigDialog.component.ts

```diff
@@ -0,0 +1,56 @@
+import "./addCustomDialog.component.scss"
+import { Component, Inject, OnInit } from "@angular/core"
+import { FormControl, Validators, AbstractControl, ValidatorFn } from "@angular/forms"
+import { CustomConfigFileStateConnector } from "../../customConfigFileStateConnector"
+import { CustomConfigHelper } from "../../../../util/customConfigHelper"
+import { filesSelector } from "../../../../state/store/files/files.selector"
+import { buildCustomConfigFromState } from "../../../../util/customConfigBuilder"
+import { State } from "../../../../state/angular-redux/state"
+
+export function createCustomConfigNameValidator(customConfigFileStateConnector: CustomConfigFileStateConnector): ValidatorFn {
+	return (control: AbstractControl): { Error: string } => {
+		const value = control.value
+		if (
+			!CustomConfigHelper.hasCustomConfigByName(
+				customConfigFileStateConnector.getMapSelectionMode(),
+				customConfigFileStateConnector.getSelectedMaps(),
+				value
+			)
+		) {
+			return null
+		}
+		return { Error: "A Custom View with this name already exists." }
+	}
+}
+
+@Component({
+	template: require("./addCustomConfigDialog.component.html")
+})
+export class AddCustomConfigDialogComponent implements OnInit {
+	private customConfigFileStateConnector: CustomConfigFileStateConnector
```

Comment:
~~this is only needed in `ngOnInit`. Therefore we can (and should :P )make it a local variable of `ngOnInit`~~

It is used in the template. So lets remove `private`